### PR TITLE
Initial work state manager

### PR DESCRIPTION
**Pull request by sharkwouter**
It doesn't do anything yet, though

### CODE REVIEW

File: src/StateManager.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include "StateManager.hpp"
+
+StateManager::StateManager() {
+    currentState = GameState();
```
Comment (protheory8) [inline src/StateManager.cpp]: I'm pretty sure line 4 is not needed.

File: src/StateManager.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+void StateManager::update() {
+    currentState.update();
```
Comment (protheory8) [inline src/StateManager.cpp]: You might want to add `this->` everywhere for consistency, but that's your choice.
```suggestion
    this->currentState.update();
```

File: src/main.cpp [hunk 3]
```diff
@@ -15,11 +16,15 @@ void handle_input(Window &window) {
 
 void run() {
     Window window("Match Theory", 800, 600);
+    StateManager stateManager = StateManager();
```
Comment (protheory8) [inline src/main.cpp]: This should work fine: 
```suggestion
    StateManager stateManager;
```

File: src/states/GameState.cpp [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#include "GameState.hpp"
+#include "../utils.hpp"
+
+GameState::GameState() : board(Board(0,0, 8,7 )){
```
Comment (protheory8) [inline src/states/GameState.cpp]:
```suggestion
GameState::GameState() : board(Board(0, 0, 8, 7)){
```

**Review by protheory8 — COMMENTED** [review]
It's a minor detail, but it would be nice if each file had a newline at the end.

**Comment by sharkwouter** [comment]
Should be good now

File: src/StateManager.cpp [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+StateManager::StateManager() {
+}
+
+void StateManager::update() {
+    this->currentState.update();
+}
+
+void StateManager::draw() {
+    currentState.draw();
```
Comment (protheory8) [inline src/StateManager.cpp]: I meant adding `this->` everywhere, including here.
```suggestion
    this->currentState.draw();
```

**Review by protheory8 — APPROVED** [review]
Thank you, LGTM!

**Pull request merged**